pkg/helper/task_runner.go

```diff
@@ -0,0 +1,52 @@
+package helper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+type task struct {
+	Name string
+	Run  func(interface{}) error
+}
+
+// TaskRunner abstracts task running engine
+type TaskRunner interface {
+	Run() error
+	// AddTask register tasks to be executed sequentially
+	// Tasks will be executed in order. First in, first to be executed.
+	AddTask(string, func(interface{}) error)
+}
+
+type taskRunnerImpl struct {
+	ctx      interface{}
```

Comment:
What does ctx mean in this case? What's the idea of this?

Reply:
see my comment https://github.com/3scale/3scale-operator/pull/357#pullrequestreview-439353338